### DISCUSSION TITLE
Se passer du consentement utilisateur

### DISCUSSION THREAD

**Discussion by JulesThePooh** [PayGreen/documentation#36] [Q&A] [answered]
Bonjour je reprend un projet que j'avais fait pour un client il y a à peu près 2 ans quand vous veniez tout juste de passer à la v3.
Apparement votre service client m'a dit qu'on pouvais complètement se passer de la checkbox à cocher quand on veut réutiliser la carte client .
Sur la documentation je vois ceci : 
![Capture d'écran 2024-09-05 130900](https://github.com/user-attachments/assets/938817d7-87f0-4d48-a4c9-31a2b9af1f6f)

Ok mais dans mon cas ça ne fonctionne absolument pas, j'ai ceci comme code : 

HTML : 
```
    <div class="row">
                    <div class="col-12 mt-5">
                        <p>{{ "cb.form.card_number.label"|trans }}</p>
                        <div id="paygreen-pan-frame" class="p-cb"></div>
                    </div>
                    <div class="col-md-6 mt-5 mb-3">
                        <p>{{ "cb.form.expiration.label"|trans }}</p>
                        <div id="paygreen-exp-frame" class="p-date"></div>
                    </div>
                    <div class="col-md-6 mt-5">
                        <p>{{ "cb.form.cvv.label"|trans }}</p>
                        <div id="paygreen-cvv-frame" class="p-cvv"></div>
                    </div>
                    <div class="col-12 mt-5">
                        <div class="w-100 d-flex justify-content-center">
                            <button id="custom-payment-button" class="p-btn-validate btn-yellow-light-standard">
                                {{ "cb.form.button.payed.label"|trans }}
                            </button>
                        </div>
                    </div>
 ```

JS : 
```
//est ce que c'est bien de cette manière qu'on utilise l'update consent ??
paygreenjs.updateConsent(true);

            paygreenjs.init({
                publicKey: publicKey,
                buyer: buyerId,
                mode: 'instrument',
                modeOptions: {
                    authorizedInstrument: true,
                },
                paymentMethod: 'bank_card',
            });

```

Jusqu'ici tout fonctionne très bien, j'ai bien ensuite un instrument

PHP 
```
            $response = $this->httpClient->request(
                'POST',
                $this->urlBase . "payment/payment-orders",
                [
                    'headers' => [
                        'Authorization' => "Bearer " . $token,
                    ],
                    'json' => [
                        "amount" => intval(round(($amount * 100))),
                        "currency" => "eur",
                        "mode" => "instant",
                        "description" => $description,
                        "reference" => $reference,
                        "merchant_initiated" => true,
                        "instrument" => $instrument,
                    ]
                ]
            );
            
```
J'essaye ensuite de prélevé la personne et c'est ici ou j'ai toujours le meme message d'erreur : {"message":"The given Instrument does not exist or is not reusable.","timestamp":"2024-09-05T11:21:18+00:00"}"

Voila, bien sur si j'enleve le updateConsent et que je l'aisse l'utilisateur cocher la checkbox, tout fonctionne correctement.
Est ce que vous pouvez me dire comment bien utilisé ce update consent ? Merci beaucoup.

**Comment by gaelDolle** [comment, marked as answer]
Bonjour,

Plusieurs pistes possibles : 

- Avez-vous bien une `<div id="paygreen-container">` non masquée dans le DOM (elle permet d'afficher la validation 3DS si besoin)
- Faites vous bien le updateConsent après le init de paygreenjs ?
- Obtenez vous bien un instrument authorized comme sur la capture ?

![image](https://github.com/user-attachments/assets/8b7d4834-9d55-49c6-8c53-c83211276d98)

**Reply by JulesThePooh** [reply]
Merci bcp c'était juste effectivement mettre le updateConsent apres le paygreenjs.init
